drivers/can/stm32_can.c

```diff
@@ -205,6 +205,8 @@ int can_stm32_runtime_configure(struct device *dev, enum can_mode mode,
 	hcan.Init.BS2  = bs2;
 	hcan.Init.Prescaler = prescaler;
 
+	hcan.State = HAL_CAN_STATE_RESET;
```

Comment:
Actual change that i made to initialize the state of the Can handle before init ( since Init checks the state before initializing)